react/features/screenshot-capture/ScreenshotCaptureSummary.tsx

```diff
@@ -121,28 +121,34 @@ export default class ScreenshotCaptureSummary {
      * @returns {void}
      */
     stop() {
-        this._streamWorker.postMessage({ id: CLEAR_INTERVAL });
+        this._streamWorker.postMessage({ id: CLEAR_TIMEOUT });
     }
 
     /**
-     * Method that is called as soon as the first frame of the video loads from stream.
-     * The method is used to store the {@code ImageData} object from the first frames
-     * in order to use it for future comparisons based on which we can process only certain
-     * screenshots.
+     * Sends to worker the imageBitmap for the next timeout.
      *
-     * @private
-     * @returns {void}
+     * @returns {Promise<void>}
      */
-    async _initScreenshotCapture() {
-        const imageBitmap = await this._imageCapture.grabFrame();
+    async sendTimeout() {
+        let imageBitmap: ImageBitmap | undefined;
 
-        this._currentCanvasContext?.drawImage(imageBitmap, 0, 0, this._streamWidth, this._streamHeight);
-        const imageData = this._currentCanvasContext?.getImageData(0, 0, this._streamWidth, this._streamHeight);
+        if (!this._imageCapture.track || this._imageCapture.track.readyState !== 'live') {
+            logger.warn('Track is in invalid state');
+            this.stop();
+
+            return;
+        }
+
+        try {
+            imageBitmap = await this._imageCapture.grabFrame();
+        } catch (e) {
+            // ignore error
```

Comment:
Do we still want to send it to the worker even when we fail?

Reply:
Yes, grabFrame throws an error quite often, but, eventually, it works, if we do not send it back to the worker as undefined when it throws the error it will stop the whole process

Reply:
sometimes this error occurs, but the track is actually in live state, and it will eventually grab a frame in the next tries.

Reply:
why won't the worker continue if we send frames later?

Reply:
Because if we do not send the message to the worker we won't receive the onmessage event back from the worker

Reply:
it is not a interval, it is a timeout